src/System.Buffers.Experimental/System/Buffers/BytesReader.cs

```diff
@@ -11,14 +11,14 @@ namespace System.Buffers
     /// </summary>
     public struct BytesReader
     {
-        readonly EncodingData _encoding;
+        readonly TextEncoder _encoding;
         ReadOnlyBytes _unreadSegments;
         int _index; // index relative to the begining of bytes passed to the constructor
 
         ReadOnlyMemory<byte> _currentSegment;
         int _currentSegmentIndex;
         
-        public BytesReader(ReadOnlyBytes bytes, EncodingData encoding)
+        public BytesReader(ReadOnlyBytes bytes, TextEncoder encoding)
```

Comment:
That's what you get for using VS refactor. :)

I'll scrub the code for all these instances of encoding and change to encoder.